Documentation/ApiOverview/FormProtection/Index.rst

```diff
@@ -17,69 +25,44 @@ request forgery (CSRF).
     protection is not supported for anonymous users. Without a logged-in user
     the token will always be :php:`dummyToken`. See :forge:`77403` for details.
 
+For each form in the backend/frontend (or link that changes some data), create a
+token and insert it as a hidden form element. The name of the form element does
+not matter; you only need it to get the form token for verifying it.
 
-.. contents::
-   :local:
-
-
-..  index:: pair; Form protection tool; Backend
-..  _csrf-backend:
+Examples
+========
 
-Usage in the backend
-====================
+..  literalinclude:: _FormProtectionExample.php
+    :caption: EXT:my_extension/Classes/Controller/FormProtectionExample.php
 
-For each form in the BE (or link that changes some data), create a token and
-insert it as a hidden form element. The name of the form element does not
-matter; you only need it to get the form token for verifying it.
+The three parameters of the :php:`generateToken()` method:
 
-..  code-block:: php
+-   :php:`$formName`
+-   :php:`$action` (optional)
+-   :php:`$formInstanceName` (optional)
 
-    // use TYPO3\CMS\Core\FormProtection\FormProtectionFactory;
-
-    $formToken = FormProtectionFactory::get()
-        ->generateToken('BE user setup', 'edit');
-    $this->content .= '<input type="hidden" name="formToken" value="' . $formToken . '">';
-
-The three parameters :php:`$formName`, :php:`$action` (optional) and
-:php:`$formInstanceName` (optional) can be arbitrary strings, but they should
-make the form token as specific as possible. For different forms (for example,
-BE user setup and editing a :sql:`tt_content` record) or different records (with
-different UIDs) from the same table, those values should be different.
+can be arbitrary strings, but they should make the form token as specific as
+possible. For different forms (for example, BE user setup and editing a
+:sql:`tt_content` record) or different records (with different UIDs) from the
+same table, those values should be different.
 
 For editing a :sql:`tt_content` record, the call could look like this:
 
 ..  code-block:: php
 
-    // use TYPO3\CMS\Core\FormProtection\FormProtectionFactory;
-
-    $formToken = FormProtectionFactory::get()
-        ->generateToken('tt_content', 'edit', $uid);
-
-Finally, you need to persist the tokens. This makes sure that
-generated tokens get saved, and also that removed tokens stay removed:
-
-..  code-block:: php
-
-    // use TYPO3\CMS\Core\FormProtection\FormProtectionFactory;
-
-    FormProtectionFactory::get()
-        ->persistTokens();
-
-In backend lists, it might be necessary to generate hundreds of tokens.
-So, the tokens are not automatically persisted after creation for performance
-reasons.
+    $formToken = $formProtection->generateToken('tt_content', 'edit', (string)$uid);
```

Comment:
So `$formProtection` is a FormProtectionFactory that got injected? Then it should be called with $this and we should show how to inject it including the class it has somewhere. Otherwise this codesnippets gets hard to understand

Reply:
Up there is, see the literalinclude. FormProtectionFactory is injected, and formProtection is retrieved from that factory in the action (or whatever is used). These snippets below are just derived from that class above. Using a full class will blow up the examples.

Reply:
Let's maybe put in the description that this is an excerpt of FormProtectionExample.php?